react/features/base/avatar/components/AbstractAvatar.js

```diff
@@ -0,0 +1,215 @@
+            colorBase
+        } = this.props;
+        const {
+            avatarFailed,
+            gravatarFailed
+        } = this.state;
+
+        const _initials = getInitials(_initialsBase);
+
+        // Precendence logic of avatars:
+
+        if (_avatarURL && _avatarURL.match(/^http[s]*:\/\//i) && !avatarFailed) {
+            // URL-based avatar first, if set and succeeds to load
+            return this._renderURLAvatar(_avatarURL);
+        } else if (_gravatarURL && !gravatarFailed) {
+            // Then gravatar if there is no URL based avatar, and there is email address
+            // and gravatar succeeds to load
+            return this._renderGravatar(_gravatarURL);
+        } else if (_initials) {
+            return this._renderInitialsAvatar(_initials, getAvatarColor(colorBase || _initialsBase));
+        }
+
+        return this._renderDefaultAvatar();
+    }
+
+    _onAvatarLoadError: () => void;
+
+    /**
+     * Callback to handle the error while loading of the avatar URI.
+     *
+     * @returns {void}
+     */
+    _onAvatarLoadError() {
+        this.setState({
+            avatarFailed: true
+        });
+    }
+
+    _onGravatarLoadError: () => void;
+
+    /**
+     * Callback to handle the error while loading of the gravatar URI.
+     *
+     * @returns {void}
+     */
+    _onGravatarLoadError() {
+        this.setState({
+            gravatarFailed: true
+        });
+    }
+
+    /**
+     * Function to render the actual, plarform specific default avatar component.
+     *
+     * @returns {React$Element<*>}
+     */
+    _renderDefaultAvatar: () => React$Element<*>
+
+    /**
+     * Function to render the actual, plarform specific Gravatar-based avatar component.
+     *
+     * @param {string} uri - The URI of the gravatar.
+     * @returns {React$Element<*>}
+     */
+    _renderGravatar: string => React$Element<*>
+
+    /**
+     * Function to render the actual, plarform specific initials-based avatar component.
+     *
+     * @param {string} initials - The initials to use.
+     * @param {string} color - The color to use.
+     * @returns {React$Element<*>}
+     */
+    _renderInitialsAvatar: (string, string) => React$Element<*>
+
+    /**
+     * Function to render the actual, plarform specific URL-based avatar component.
+     *
+     * @param {string} uri - The URI of the avatar.
+     * @returns {React$Element<*>}
+     */
+    _renderURLAvatar: string => React$Element<*>
+}
+
+/**
+ * Maps part of the Redux state to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {Props} ownProps - The own props of the component.
+ * @returns {Props}
+ */
+export function _mapStateToProps(state: Object, ownProps: Props) {
+    const { participantId } = ownProps;
+    const _participant = participantId && getParticipantById(state, participantId);
+    const _initialsBase = (_participant && (_participant.name || _participant.email)) || ownProps.displayName;
```

Comment:
I think we shouldn't use the email as initials.

Reply:
Nit: you can deconstruct displayName at the top toghether with participantId.

Reply:
I added the deconstruct, thanks.

Why do you think we shouldn't use the email for initials?